### ISSUE TITLE
Tensorflow_datasets version

### ISSUE THREAD

**Issue by daniel-trejobanos** [BorgwardtLab/Set_Functions_for_Time_Series#5]
Hi, 

If I install Seft using poetry, the environment contains tensorflow 1.15 and tensorflow_datasets 4.2, thus when I try to generate my own data set using the generated environment, I run into a dependency error (as tfds 4.2 requires tf 2.1).

Maybe the solution is to add tfds == 2.0 explicitly in SEFT pyproject.toml file?

Thank you and best regards.

**Comment by daniel-trejobanos** [comment]
Hi, I will build upon my previous post. I have a to deploy within a container, and the container is run in a server without outside network access, thus installing anything requires a big overhead (build container and transfer it to the server). If I want a reproducible workflow, I would like to install all dependencies within the container build process.

Now on to the issue. As I mentioned before, if I let poetry choose the tfds version I cannot create my own class and load it using tfds.load() as I get an error of wrong  tensorflow version. If I use tensorflow 2 I can load my data set without problems, I have tested it already. 

 If I install a compatible version of tfds (2.1.0)  I cannot build my dataset as it wishes to connect to the internet to check if the dataset is in gcs. I have also tried using a dataset generated by tfds 4.2.0 and read it using tfds 2.1.0 but I get errors again:
 ' TypeError: Unsupported return value from function passed to Dataset.map(): {'combined': (None, <tf.Tensor 'args_2:0' shape=(?,) dtype=float32>, <tf.Tensor 'concat/concat:0' shape=(?, 7) dtype=float32>, <tf.Tensor 'IsFinite:0' shape=(?, 7) dtype=bool>, <tf.Tensor 'strided_slice:0' shape=() dtype=int32>), 'target': <tf.Tensor 'args_1:0' shape=(?, 1) dtype=int32>}.'.

To reiterate my data is in a secure network without internet access and I am not allowedto move it thus, I can only build the dataset on the server without connection.

 I wonder if you had the same issues and how did you manage to overcome them?

p.s. Have you tried porting your code to tensorflow 2.0 and python3.8 ?

Thank you and best regards.
